Wikipedia/Code/TalkPageViewController.swift

```diff
@@ -194,6 +202,33 @@ class TalkPageViewController: ViewController {
         push(coffeeViewController, animated: true)
     }
 
+    @objc private func userDidTapChangeLanguage() {
+        if viewModel.pageType == .user {
+            let languageVC = WMFPreferredLanguagesViewController.preferredLanguagesViewController()
+            languageVC.delegate = self
+
+            if let themeableVC = languageVC as Themeable? {
+                themeableVC.apply(theme: self.theme)
+            }
+            present(WMFThemeableNavigationController(rootViewController: languageVC, theme: self.theme), animated: true, completion: nil)
+
+        } else if viewModel.pageType == .article {
+            guard let languageCode  = viewModel.siteURL.wmf_contentLanguageCode else {
+                return
+            }
+            if let articleTitle = viewModel.pageTitle.extractingArticleTitleFromTalkPage(lang: languageCode).denormalizedPageTitle {
+                var talkPageURLComponents = URLComponents(url: viewModel.siteURL, resolvingAgainstBaseURL: false)
+                talkPageURLComponents?.path = "/wiki/\(articleTitle)"
+                if let urlValue = talkPageURLComponents?.url {
+                    let languageVC = WMFArticleLanguagesViewController(articleURL: urlValue)
```

Comment:
I did a little bit of testing and this works better (under-the-hood it assigns a `languageVariantCode` that is needed for ZH variants to appear in the list):

```
if let articleTitle = viewModel.pageTitle.extractingArticleTitleFromTalkPage(lang: languageCode).denormalizedPageTitle,
    let articleURL = viewModel.siteURL.wmf_URL(withTitle: articleTitle) {
        let languageVC = WMFArticleLanguagesViewController(articleURL: articleURL)
```